### PR TITLE
feat(api): Implement endpoints to create/delete org level saved searches (SEN-399)

### PR DESCRIPTION
**Pull request by wedamija**
Allows org owners and admins to create/delete org wide saved searches

### CODE REVIEW

File: src/sentry/api/endpoints/organization_searches.py [hunk 1]
```diff
@@ -75,3 +86,28 @@ def get(self, request, organization):
...
+
+        if serializer.is_valid():
+            result = serializer.object
+            # Prevent from creating duplicate queries
+            if SavedSearch.objects.filter(
+                Q(is_global=True) | Q(organization=organization, owner__isnull=True),
+                query=result['query'],
+            ).exists():
+                return Response(
+                    {'detail': 'Query {} already exists'.format(result['query'])},
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_searches.py]:
```suggestion
                    {'detail': u'Query {} already exists'.format(result['query'])},
```

Potential UnicodeError :ghost:

File: src/sentry/api/endpoints/organization_searches.py [hunk 2]
```diff
@@ -13,7 +17,14 @@
 from sentry.models.search_common import SearchType
 
 
+class OrganizationSearchSerializer(serializers.Serializer):
+    type = serializers.IntegerField(required=True)
+    name = serializers.CharField(required=True)
+    query = serializers.CharField(required=True)
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_searches.py]: Do we want to ensure that the query has some bytes? Is it useful to save empty queries?

File: src/sentry/api/endpoints/organization_search_details.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        Delete a saved search
+
+        Permanently remove a saved search.
+
+            {method} {path}
+
+        """
+        try:
+            search = SavedSearch.objects.get(
+                owner__isnull=True,
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_search_details.py]: Are we going to add deleting pinned searches later? Or in separate endpoints?

File: src/sentry/api/endpoints/organization_searches.py (hunk 2 above)
Comment (wedamija) [reply src/sentry/api/endpoints/organization_searches.py]: Yeah, probably worthwhile

File: src/sentry/api/endpoints/organization_search_details.py (hunk 3 above)
Comment (wedamija) [reply src/sentry/api/endpoints/organization_search_details.py]: Yeah, separate pr incoming when I can figure out why github is rejecting it 🤷‍♂️

**Comment by wedamija** [comment]
@markstory Could you take a look at this? We're ready to land now since the DDL has gone through.

**Review by markstory — APPROVED** [review]
Looks good to me. Nice work on the tests.

**Pull request merged**